include/deal.II/lac/solver_gmres.h

```diff
@@ -1059,171 +1338,377 @@ namespace internal
 
 
 
-    /**
-     * Orthogonalize the vector @p vv against the @p dim (orthogonal) vectors
-     * given by @p orthogonal_vectors using the modified or classical
-     * Gram-Schmidt algorithm.
-     * The factors used for orthogonalization are stored in @p h. The boolean @p
-     * re_orthogonalize specifies whether the Gram-Schmidt algorithm
-     * should be applied twice. The algorithm checks loss of orthogonality in
-     * the procedure every fifth step and sets the flag to true in that case.
-     * All subsequent iterations use re-orthogonalization.
-     * Calls the signal re_orthogonalize_signal if it is connected.
-     */
+    inline void
+    ArnoldiProcess::initialize(
+      const LinearAlgebra::OrthogonalizationStrategy orthogonalization_strategy,
+      const unsigned int                             basis_size,
+      const bool                                     force_reorthogonalization)
+    {
+      this->orthogonalization_strategy = orthogonalization_strategy;
+      this->do_reorthogonalization     = force_reorthogonalization;
+
+      hessenberg_matrix.reinit(basis_size + 1, basis_size);
+      triangular_matrix.reinit(basis_size + 1, basis_size, true);
+
+      // some additional vectors, also used in the orthogonalization
+      projected_rhs.reinit(basis_size + 1, true);
+      givens_rotations.reserve(basis_size);
+
+      if (orthogonalization_strategy ==
+          LinearAlgebra::OrthogonalizationStrategy::
+            delayed_classical_gram_schmidt)
+        h.reinit(2 * basis_size + 3);
+      else
+        h.reinit(basis_size + 1);
+    }
+
+
+
     template <typename VectorType>
     inline double
-    iterated_gram_schmidt(
-      const LinearAlgebra::OrthogonalizationStrategy orthogonalization_strategy,
-      const internal::SolverGMRESImplementation::TmpVectors<VectorType>
-                                               &orthogonal_vectors,
+    ArnoldiProcess::orthonormalize_nth_vector(
```

Comment:
Maybe rename `dim`? I guess it is `n`?

Reply:
We have many occurrences; I took the chance to change them all from `dim` (which we use with different connotation in most places of deal.II, so I agree with your assessment) to `n`. The one downside is that we have many uses of `n` formerly `dim`, so that might be a question. I left it as a separate comment, please express your opinion to see what we like more.